frontend/src/store/index.js

```diff
@@ -285,6 +289,14 @@ function firstItemMatchingVersionClassification (items) {
   return head(items)
 }
 
+function filterShortcuts ({ getters, shortcuts, targetsFilter }) {
+  shortcuts = filter(shortcuts, ({ target }) => (target === TargetEnum.CONTROL_PLANE && getters.hasControlPlaneTerminalAccess) || target !== TargetEnum.CONTROL_PLANE)
+  shortcuts = filter(shortcuts, ({ target }) => (target === TargetEnum.GARDEN && getters.hasGardenTerminalAccess) || target !== TargetEnum.GARDEN)
+  shortcuts = filter(shortcuts, ({ target }) => ((target === TargetEnum.SHOOT && getters.hasShootTerminalAccess) || target !== TargetEnum.SHOOT))
+  shortcuts = filter(shortcuts, ({ target }) => includes(targetsFilter, target))
+  return shortcuts
+}
+
```

Comment:
```suggestion
function filterShortcuts (state, getters) {
  return (shortcuts, targets) => {
    shortcuts = filter(shortcuts, ({ target }) => (target === TargetEnum.CONTROL_PLANE && getters.hasControlPlaneTerminalAccess) || target !== TargetEnum.CONTROL_PLANE)
    shortcuts = filter(shortcuts, ({ target }) => (target === TargetEnum.GARDEN && getters.hasGardenTerminalAccess) || target !== TargetEnum.GARDEN)
    shortcuts = filter(shortcuts, ({ target }) => ((target === TargetEnum.SHOOT && getters.hasShootTerminalAccess) || target !== TargetEnum.SHOOT))
    shortcuts = filter(shortcuts, ({ target }) => includes(targets, target))
    return shortcuts
  }
}
```
Use getters syntax or actions syntax or separate context from other parameters.